### PR TITLE
[Merged by Bors] - refactor(RingHom.injective): use simple ring

### PR DESCRIPTION
**Pull request by jjaassoonn**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/ENat/Lattice.lean [hunk 1]
```diff
@@ -190,11 +190,11 @@ lemma iSup_add_iSup_of_monotone {ι : Type*} [Preorder ι] [IsDirected ι (· 
...
 
-proof_wanted smul_iSup {R} [SMul R ℕ∞] [IsScalarTower R ℕ∞ ℕ∞] (f : ι → ℕ∞) (c : R) :
-    c • ⨆ i, f i = ⨆ i, c • f i
+-- proof_wanted smul_iSup {R} [SMul R ℕ∞] [IsScalarTower R ℕ∞ ℕ∞] (f : ι → ℕ∞) (c : R) :
+--     c • ⨆ i, f i = ⨆ i, c • f i
 
-proof_wanted smul_sSup {R} [SMul R ℕ∞] [IsScalarTower R ℕ∞ ℕ∞] (s : Set ℕ∞) (c : R) :
-    c • sSup s = ⨆ a ∈ s, c • a
+-- proof_wanted smul_sSup {R} [SMul R ℕ∞] [IsScalarTower R ℕ∞ ℕ∞] (s : Set ℕ∞) (c : R) :
+--     c • sSup s = ⨆ a ∈ s, c • a
```
Comment (jjaassoonn) [inline Mathlib/Data/ENat/Lattice.lean:198]: It seems `lake shake` does not look at `proof_wanted` and does not want me to add imports which makes `IsScalarTower` available in this file

Comment (jcommelin) [reply Mathlib/Data/ENat/Lattice.lean:198]: Please add an entry to `scripts/noshake.json` to keep the import anyways.

Also, please report this problem as a github issue: we should fix shake.

File: Mathlib/FieldTheory/Galois/GaloisClosure.lean [hunk 2]
```diff
@@ -58,13 +58,13 @@ instance (L₁ L₂ : IntermediateField k K) [IsGalois k L₁] [IsGalois k L₂]
 
 instance (L₁ L₂ : IntermediateField k K) [FiniteDimensional k L₁] :
     FiniteDimensional k ↑(L₁ ⊓ L₂) :=
-  .of_injective (IntermediateField.inclusion inf_le_left).toLinearMap
-    (IntermediateField.inclusion inf_le_left).injective
+  .of_injective (IntermediateField.inclusion (E := L₁ ⊓ L₂) (F := L₁) inf_le_left).toLinearMap
```
Comment (jcommelin) [inline Mathlib/FieldTheory/Galois/GaloisClosure.lean:61]: Do you understand why we need to help the elaborator more now?

File: Mathlib/Data/ENat/Lattice.lean (hunk 1 above)
Comment (jjaassoonn) [reply Mathlib/Data/ENat/Lattice.lean]: I need to double check my claim that `shake` does not look at `proof_wanted`, I cannot seem to reproduce the behaviour

**Comment by jcommelin** [comment]
This seems like the right thing to do mathematically. The minor increase in imports is worth it.

bors d+

**Pull request closed without merging**